h5netcdf/tests/test_h5netcdf.py

```diff
@@ -1137,3 +1137,63 @@ def test_overwrite_existing_file(tmp_local_netcdf):
     # check attribute
     with h5netcdf.File(tmp_local_netcdf, "r") as ds:
         assert ds.attrs._h5attrs.get("_NCProperties", False)
+
+
+def test_create_attach_scales(tmp_local_netcdf):
+    # create file with netCDF4
+    with netCDF4.Dataset(tmp_local_netcdf, "w") as ds:
+        ds.createDimension("x", 0)
+        ds.createDimension("y", 1)
+        ds.createVariable("test", "i4", ("x",))
+        ds.variables["test"] = np.ones((10,))
+
+    # append file with netCDF4
+    with netCDF4.Dataset(tmp_local_netcdf, "a") as ds:
+        ds.createVariable("test1", "i4", ("x",))
+        ds.createVariable("y", "i4", ("x", "y"))
+
+    # check scales
+    with h5netcdf.File(tmp_local_netcdf, "r") as ds:
+        refs = ds._h5group["x"].attrs.get("REFERENCE_LIST", False)
+        assert len(refs) == 3
+        for (ref, dim), name in zip(refs, ["/test", "/test1", "/_nc4_non_coord_y"]):
+            assert dim == 0
+            assert ds._root._h5file[ref].name == name
+
+    # create file with netCDF4
```

Comment:
These lines look duplicated from above?

Reply:
Yes, correct, I recreate the file to append with legacyapi. I could create a special function to reduce the duplication.

Reply:
I've de-duplicated the test code. Thanks @shoyer.